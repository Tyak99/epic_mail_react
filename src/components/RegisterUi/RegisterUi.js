import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import FormGroup from '../FormGroup/FormGroup';
import style from './registerUi.css';

const RegisterUi = ({
  handleInput, submit, error, redirect, loading,
}) => {
  const formProps = [
    { type: 'text', placeholder: 'First Name', id: 'firstName' },
    { type: 'text', placeholder: 'Last Name', id: 'lastName' },
    { type: 'email', placeholder: 'Email Address', id: 'email' },
    { type: 'password', placeholder: 'Password', id: 'password' },
    {
      type: 'password',
      placeholder: 'Confirm Password',
      id: 'confirmPassword',
    },
  ];
  return (
    <div className={style.container}>
      {redirect}
      <div className={style.backgroundLogin}>
        <img
          src="https://directlinedev.com/media/services/service/background/background_11.wide.jpeg"
          alt=""
        />
      </div>
      <div className={style.formSection}>
        <form onSubmit={submit}>
          <h2 className="formTitle">Sign Up</h2>
          <p className={style.error}>
            { error }
          </p>
          {formProps.map(prop => (
            <FormGroup
              type={prop.type}
              placeholder={prop.placeholder}
              id={prop.id}
              key={prop.id}
              handleInput={handleInput}
              name={prop.id}
            />
          ))}
          <div style={{ textAlign: 'center' }}>
            <button id="submit" type="submit" className={style.submitBtn}>
              {loading ? 'Loading...' : 'Register'}
            </button>
          </div>
          <p className={style.signInLink}>
            Already have an account?
            <Link to="/">
              <span>
                Sign In
              </span>
            </Link>
          </p>
        </form>
      </div>
    </div>
  );
};

RegisterUi.defaultProps = {
  redirect: null,
  error: null,
  loading: false,
};
RegisterUi.propTypes = {
  handleInput: PropTypes.func.isRequired,
  submit: PropTypes.func.isRequired,
  error: PropTypes.string,
  redirect: PropTypes.element,
  loading: PropTypes.bool,
};
export default RegisterUi;
