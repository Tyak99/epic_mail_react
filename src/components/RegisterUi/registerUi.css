.container {
    display: flex;
    flex-wrap: wrap;
    height: 100vh;
  }
  
  .backgroundLogin {
    width: calc(100% - 520px);
    background: linear-gradient(to right, #0083b0, #00b4db);
  }
  .backgroundLogin img {
    width: 100%;
    height: 100vh;
  }
  .error {
      color: crimson;
      font-size: 1.2rem;
  }
  .formSection {
    width: 520px;
  }
  .formSection form {
    max-width: 323px;
    padding: 30px;
    margin: 10% auto;
  }
  
  .submitBtn {
    padding: 10px 40px;
    background-color: #49738c;
    color: white;
    cursor: pointer;
    border-radius: 0.5em;
    font-size: 18px;
    font-weight: 500;
  }
  .submitBtn:hover {
    background-color: #db5756;
  }
  
  .signInLink {
    text-align: center;
  }
  .signInLink a {
    text-decoration: none;
    margin-left: 8px;
    color: #db5756;
  }
  .signInLink a:hover {
      color: crimson;
  }
