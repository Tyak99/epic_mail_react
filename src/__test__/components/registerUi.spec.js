import React from 'react';
import { shallow } from 'enzyme';
import RegisterUi from '../../components/RegisterUi/RegisterUi';

const props = {
  submit: () => {},
  handleInput: () => {},
};
describe('component: RegisterUi', () => {
  it('should render correctly', (done) => {
    const wrapper = shallow(<RegisterUi {...props} />);
    expect(wrapper).toMatchSnapshot();
    done();
  });
});
