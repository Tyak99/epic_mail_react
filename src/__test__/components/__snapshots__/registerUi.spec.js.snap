// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: RegisterUi should render correctly 1`] = `ShallowWrapper {}`;
